src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -189,7 +189,7 @@ public DebuggedProcess(bool bLaunched, LaunchOptions launchOptions, ISampleEngin
                 // For local Linux and OS X launch, use the local Unix transport which creates a new terminal and
                 // uses fifos for debugger (e.g., gdb) communication.
                 if (this.MICommandFactory.UseExternalConsoleForLocalLaunch(localLaunchOptions) &&
-                    (PlatformUtilities.IsLinux() || PlatformUtilities.IsOSX())
+                    (PlatformUtilities.IsLinux() || (PlatformUtilities.IsOSX() && localLaunchOptions.DebuggerMIMode == MIMode.Gdb))
```

Comment:
This is used for clrdbg as well. I think you want

`localLaunchOptions.DebuggerMIMode != MIMode.Lldb`